lustrefs-exporter/src/lib.rs

```diff
@@ -17,27 +19,76 @@ use axum::{
     response::{IntoResponse, Response},
 };
 use lustre_collector::{LustreCollectorError, TargetVariant};
-use opentelemetry_sdk::{Resource, metrics::SdkMeterProvider};
-use prometheus::Registry;
+use prometheus_client::{encoding::EncodeMetric, metrics::TypedMetric, registry::Registry};
+
+pub type LabelContainer = Vec<(&'static str, String)>;
+pub type PrometheusFamily<T> = prometheus_client::metrics::family::Family<LabelContainer, T>;
+
+/// A wrapper around prometheus_client::metrics::family::Family. It contains an `is_set`
+/// field that is set to true when `get_or_create` is called. Since registering a
+/// family can be costly, we can avoid registering families multiple times.
+#[derive(Default, Debug)]
+pub struct Family<T> {
+    inner: PrometheusFamily<T>,
+    is_set: bool,
+}
```

Comment:
By blocking the constructor generic (`C` in `Family<S, M, C>`) we block access to the constructor if we ever need to construct histograms. Ask me how I know. A more inclusive definition should be:
```
/// A wrapper around prometheus_client::[...]::Family that includes a name field
#[derive(Debug)]
pub struct Family<T, C = fn() -> T> {
    inner: PrometheusFamily<LabelContainer, T, C>,
    is_set: bool,
}
```